packages/server/src/api/controllers/permission.ts

```diff
@@ -79,7 +79,7 @@ async function updatePermissionOnRole(
     ) {
       rolePermissions[resourceId] =
         typeof rolePermissions[resourceId] === "string"
-          ? [rolePermissions[resourceId]]
+          ? [rolePermissions[resourceId] as unknown as string]
```

Comment:
This was a weird one. The type of `rolePermissions[resourceId]` is `string[]` according to `Role.permissions` but I didn't want to remove this explicit check in case it's here for older apps that might have different data to what we expect.

Reply:
This is likely to be what is going on here, its probably something to do with old apps that may have had a slightly different data strategy - we could comment it for removal at some point in the future/if anyone comes across it and has a problem with it they are aware its not critical/linked to something very old/likely no longer needed.